sqlmesh/core/test/definition.py

```diff
@@ -548,7 +555,8 @@ def runTest(self) -> None:
 
     def _render_model_query(self) -> exp.Query:
         variables = self.body.get("vars", {}).copy()
-        time_kwargs = {key: variables.pop(key, None) for key in TIME_KWARG_KEYS}
+        variables_keys = variables.copy().keys()
```

Comment:
you probably don't need the keys() part for x in dict is the same right?

Reply:
why copy this at all? can you just do for key in TIME_KWARG_KEYS?

Reply:
If we do `{key: variables.pop(key, None) for key in TIME_KWARG_KEYS}`, then every macro variable in it that is NOT overridden gets assigned a value of `None` that's passed to the renderer. 

If a model query uses time variables that were not overridden, they then render with `None` instead of whatever value they would normally have.

Reply:
> you probably don't need the keys() part for x in dict is the same right?

yes, fixed

Reply:
should be in {key: variables.pop(key) for key in TIME if key in variables}

Reply:
Ahhh - thank you. 

I initially did something similar with the for/if reversed. It hit this error, which I incorrectly assumed would happen if I conditioned directly on `variables` in any way.

```
    time_kwargs = {key: variables.pop(key) for key in variables if key in TIME_KWARG_KEYS}
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: dictionary changed size during iteration
```